src/Core/Scheduler/Providers/TileProvider.js

```diff
@@ -49,47 +42,78 @@ TileProvider.prototype.preprocessDataLayer = function preprocessLayer(layer, vie
     });
 };
 
+const worldQuaternion = new THREE.Quaternion();
 TileProvider.prototype.executeCommand = function executeCommand(command) {
-    var extent = command.extent;
+    const extent = command.extent;
     if (command.requester &&
         !command.requester.material) {
         // request has been deleted
         return Promise.reject(new CancelledCommandException(command));
     }
+    const layer = command.layer;
+    const builder = layer.builder;
+    const parent = command.requester;
+    const level = (command.level === undefined) ? (parent.level + 1) : command.level;
 
-    var parent = command.requester;
+    if (!this.cacheGeometry[level]) {
+        this.cacheGeometry[level] = new Map();
+    }
 
+    const ce = builder.getCommonGeometryExtent(extent);
```

Comment:
I don't think TileProvider needs to be aware of this: this could an internal feature of each tile builder.

But even without this, the API  could be simplified a lot if `getCommonGeometryExtent` was something that would return 2 things: the new extent and the transformation to apply.
And this would give you a better naming as well: something like: `computeSharableExtent`.

WDYT?